odoo/addons/base/models/ir_module.py

```diff
@@ -244,6 +244,13 @@ def _get_icon_image(self):
                 with tools.file_open(path, 'rb') as image_file:
                     module.icon_image = base64.b64encode(image_file.read())
 
+    def _get_i18n_location(self):
+        # TODO fallback on config settings?
+        default = "https://nightly.odoo.com"
+        for module in self:
+            manifest_info = odoo.http.addons_manifest.get(module.name, {})
+            module.i18n_location = manifest_info.get('i18n_location', default)
```

Comment:
@naglis even non-html related, translations can be sensitive so it is a good question. On the one hand I want to be able for custom modules to provide translations this way, on the other hand, how to ensure it is safe? The manifest info shouldn't be editable by users so we can rely on I guess. What risk are you trying to prevent with a digital signature? If you add pgp signature, you need to handle trust of the key which is key of the same problem.

Reply:
@mart-e, well, yeah, one could also argue that now it is even easier, e.g. an attacker could simply introduce malicious translations on Transifex.

I am not competent enough to do risk assessment on this, so that's why I've asked @odony :)